File: Java/08-Laço While/Fluxo.java
class Fluxo {
    public static void main(String[] args) {
        // Vamos interromper o fluxo do laço usando continue
        int contador = 0;

        // Enquanto o contador for menor ou igual 10 ele vai incrementar
        while (contador < 10) {
            contador++;
            // Quando o contador valer 5 ou 8 ele vai ignorar o que esta abaixo do continue
            // e vai para a proxima iteração (5 e 8 não serão exibidos)
            if (contador == 5 || contador == 8)
                continue;
            System.out.println(contador);
        }

        System.out.println("Contagem finalizada!");

        System.out.println("Contagem 2");

        // Ja o break sai do laço e vai para o bloco de comandos seguinte
        int cont = 0;
        while (cont < 10) {
            System.out.println(cont);
            cont++;
            if (cont == 6)
                break; // Se cont vale 6 vai sair imediatamente deste laço e vai imprimir até 5
        }
        System.out.println("Contagem interrompida!");
    }
}

File: C/10-Laço While/Fluxo.c
#include<stdio.h>

int main(void)
{
    //podemos alterar o fluxo de um laço com alguns comandos
    int cont = 0;

    while(cont < 10) {
        cont++;
        //Variavel vai incrementar em 1 a cada iteração

        if(cont == 4 || cont == 7) {
        //Este comando volta imediatamente para o teste ignorando instruções abaixo dele
            continue;
        }
        //4 e 7 não serão impressos pois continue impedira programa de chegar nesta instrução
        //voltando ao inicio do while imediatamente
        printf("%d ", cont);
    }
    printf("\nContagem finalizada \n");

    //Tambem temos o break que encerra imedatamente o laço se for acionado
    while(cont > 0) { //contador deve ir até 0
        cont--;

        //break sera acionado quando cont valer 5 interrompendo este laço imediatamente
        if(cont == 5) { //5 não será impresso pois impressão está abaixo do break
            break;
        }
        printf("%d \n",cont);
    }
    printf("Contagem interrompida");
    return 0;
}
